### PR TITLE
Fixing race in setting timeout of the Remoter object [#1097]

### PR DESCRIPTION
**Pull request by bentsi**
## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I gave variables/functions meaningful self-explanatory names
- [ ] I didn't leave commented-out/debugging code
- [ ] I didn't copy-paste code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (`hydra unit-tests`)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/remote.py [hunk 1]
```diff
@@ -53,17 +57,27 @@ class ConnectionCmdTimeout(Connection):
...
 
     def run(self, command, **kwargs):
+        self.timeout_lock.acquire()
         self.cmd_timeout = kwargs.pop('cmd_timeout', None)
-        return super(ConnectionCmdTimeout, self).run(command, **kwargs)
+        try:
+            return super(ConnectionCmdTimeout, self).run(command, **kwargs)
+        except:
+            if not self.timeout_lock.locked():
+                self.timeout_lock.release()
```
Comment (aleksbykov) [inline sdcm/remote.py]: i think Exception should be reraised, otherwise, it looks, it will not catch

File: sdcm/remote.py [hunk 2]
```diff
@@ -53,17 +57,27 @@ class ConnectionCmdTimeout(Connection):
...
+        self.config._set(timeout_lock=Lock())
 
     def run(self, command, **kwargs):
+        self.timeout_lock.acquire()
         self.cmd_timeout = kwargs.pop('cmd_timeout', None)
-        return super(ConnectionCmdTimeout, self).run(command, **kwargs)
+        try:
+            return super(ConnectionCmdTimeout, self).run(command, **kwargs)
+        except:
+            if not self.timeout_lock.locked():
```
Comment (aleksbykov) [inline sdcm/remote.py]: is method Lock.locked() exists?

**Comment by fruch** [comment]
Might be a better fix:
https://github.com/pyinvoke/invoke/pull/645
and
https://github.com/fabric/fabric/pull/1989

**Comment by bentsi** [comment]
closing this, since @fruch is adding a PR to invoke/fabric 
https://github.com/pyinvoke/invoke/pull/645#pullrequestreview-256772929

**Pull request closed without merging**